packages/offix-datastore/src/storage/adapters/WebSQLAdapter.ts

```diff
@@ -0,0 +1,250 @@
+import "websql";
+import invariant from "tiny-invariant";
+import { StorageAdapter } from "../api/StorageAdapter";
+import { PredicateFunction, ModelFieldPredicate } from "../../predicates";
+import { generateId } from "../LocalStorage";
+import { createLogger } from "../../utils/logger";
+import { ModelSchema } from "../../ModelSchema";
+
+const logger = createLogger("sqlite");
+
+const errorCallback = (tx: SQLTransaction, err: SQLError) => {
+  logger("error", err);
+  return false;
+};
+
+/**
+ * Web Storage Implementation for DataStore using IndexedDB
+ */
+export class WebSQLAdapter implements StorageAdapter {
+  private sqlite: Database;
+  private stores: ModelSchema[] = [];
+  private dbName: string;
+  private schemaVersion: number;
+
+  constructor(dbName: string, schemaVersion: number) {
+    this.dbName = dbName;
+    this.schemaVersion = schemaVersion;
+    invariant("openDatabase" in window, "WebSQL not supported");
+    this.sqlite = window.openDatabase(
+      dbName,
+      schemaVersion.toString(),
+      "Offix datastore", 5 * 1024 * 1024
+    );
+  }
+  // TODO Wrong architecture. Store can be created on demand
+  public addStore(config: ModelSchema) {
+    this.stores.push(config);
+  }
+
+  public async createStores() {
+    const existingStoreNames = await this.getStoreNames();
+    existingStoreNames.forEach((storeName) => {
+      const existingModelStoreName = this.stores.find(((store) => (storeName === store.getName())));
+      if (existingModelStoreName) { return; }
+      this.sqlite.transaction((tx) => {
+        tx.executeSql("DROP TABLE ?", [existingModelStoreName], () => {
+          logger("Store deleted", existingModelStoreName);
+        }, errorCallback);
+      });
+    });
+
+    this.stores.forEach((store) => {
+      if (existingStoreNames.includes(store.getName())) { return; }
+      const stmt = this.getCreateStatement(store);
+      this.sqlite.transaction((tx) => {
+        tx.executeSql(stmt, [], () => logger("Store created", store), errorCallback);
+      });
+    });
+  }
+
+  public async save(storeName: string, input: any): Promise<any> {
+    const [cols, vals] = prepareStatement(input, "insert");
+    const stmt = `INSERT INTO ${storeName} ${cols}`;
+    return this.transaction(stmt, vals);
+  }
+
+  public async query(storeName: string, predicate?: PredicateFunction): Promise<any> {
+    if (!predicate) {
+      return await this.getStore(storeName);
```

Comment:
I will rename it to `fetchAll`

Reply:
@wtrocki I think this is what it might be called in the interface, but I can rename it and update in the `indexeddb` adapter too

Reply:
Nope, I was wrong. Changing it now